### ISSUE TITLE
Pycharm exception with ver 1.15.0

### ISSUE THREAD

**Issue by nick4u** [tonybaloney/pycharm-security#105]
Pycharm reports at start exception in "Python Security" plugin (ver 1.15.0)

Exception :
```
java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
	at java.base/java.lang.String.substring(String.java:1874)
	at security.packaging.BasePackageChecker.parseVersionSpec(PackageChecker.kt:43)
	at security.packaging.BasePackageChecker.access$parseVersionSpec(PackageChecker.kt:20)
	at security.packaging.BasePackageChecker$parseVersionSpecs$2.apply(PackageChecker.kt:37)
	at security.packaging.BasePackageChecker$parseVersionSpecs$2.apply(PackageChecker.kt:20)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at security.packaging.BasePackageChecker.parseVersionSpecs(PackageChecker.kt:38)
	at security.packaging.SafetyDbChecker.hasMatch(SafetyDbChecker.kt:71)
	at security.packaging.PyPackageSecurityScan.inspectLocalPackages(PyPackageSecurityScan.kt:59)
	at security.packaging.PyPackageSecurityScan.checkPackagesInSdks(PyPackageSecurityScan.kt:48)
	at security.packaging.PyPackageSecurityScan.checkPackages(PyPackageSecurityScan.kt:29)
	at security.packaging.PythonPackageVulnerabilityStartupTask$runActivity$1.run(PythonPackageVulnerabilityStartupTask.kt:16)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:888)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.lambda$run$0(CoreProgressManager.java:402)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:221)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:402)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**To Reproduce**
Code or Steps to reproduce the behavior:
1. Start Pycharm

**Additional context**
 - Plugin Version: 1.15.0
 - Python Version: 3.8.0
 - Pycharm version:
```
PyCharm 2019.3.4 (Professional Edition)
Build #PY-193.6911.25, built on March 18, 2020
Runtime version: 11.0.6+8-b520.43 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 5.3.0-45-generic
GC: ParNew, ConcurrentMarkSweep
Memory: 1942M
Registry: 
Non-Bundled Plugins: GrepConsole, Statistic, com.dropbox.plugins.mypy_plugin, com.dubreuia, com.intellij.ideolog, com.intellij.lang.jsgraphql, com.jetbrains.grayTheme, com.koxudaxi.pydantic, ideanginx9, name.kropp.intellij.makefile, org.toml.lang, org.rust.lang, org.tonybaloney.security.pycharm-security, ru.adelf.idea.dotenv, ru.meanmail.plugin.requirements
```

**Comment by ajcsimons** [comment]
I too get the same stacktrace on startup:

`java.lang.StringIndexOutOfBoundsException: begin 0, end 3, length 2
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:3319)
	at java.base/java.lang.String.substring(String.java:1874)
	at security.packaging.BasePackageChecker.parseVersionSpec(PackageChecker.kt:43)
	at security.packaging.BasePackageChecker.access$parseVersionSpec(PackageChecker.kt:20)
	at security.packaging.BasePackageChecker$parseVersionSpecs$2.apply(PackageChecker.kt:37)
	at security.packaging.BasePackageChecker$parseVersionSpecs$2.apply(PackageChecker.kt:20)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at security.packaging.BasePackageChecker.parseVersionSpecs(PackageChecker.kt:38)
	at security.packaging.SafetyDbChecker.hasMatch(SafetyDbChecker.kt:71)
	at security.packaging.PyPackageSecurityScan.inspectLocalPackages(PyPackageSecurityScan.kt:59)
	at security.packaging.PyPackageSecurityScan.checkPackagesInSdks(PyPackageSecurityScan.kt:48)
	at security.packaging.PyPackageSecurityScan.checkPackages(PyPackageSecurityScan.kt:29)
	at security.packaging.PythonPackageVulnerabilityStartupTask$runActivity$1.run(PythonPackageVulnerabilityStartupTask.kt:16)
	at com.intellij.openapi.progress.impl.CoreProgressManager$TaskRunnable.run(CoreProgressManager.java:888)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:163)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:585)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:531)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:59)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:150)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.lambda$run$0(CoreProgressManager.java:402)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:221)
	at com.intellij.openapi.progress.impl.CoreProgressManager$4.run(CoreProgressManager.java:402)
	at com.intellij.openapi.application.impl.ApplicationImpl$1.run(ApplicationImpl.java:238)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)`

Plugin v 1.15.0

PyCharm 2019.3.4 (Professional Edition)
Build #PY-193.6911.25, built on March 18, 2020
Licensed to Bubble Limited / Andrew Simons
Subscription is active until November 26, 2020
Runtime version: 11.0.6+8-b520.43 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.13.1
GC: ParNew, ConcurrentMarkSweep
Memory: 1378M
Cores: 8
Registry: 
Non-Bundled Plugins: org.tonybaloney.security.pycharm-security, ru.meanmail.plugin.requirements

**Comment by tonybaloney** [comment]
Fixed in 1.15.1.

Sorry!
